File: src/main/java/ar/edu/um/tesoreria/rest/model/ChequeraImpresionDetalle.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.OffsetDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(name = "chequera_d_impresion")
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class ChequeraImpresionDetalle extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 7700613846583473862L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long chequeraImpresionDetalleId;

	private Long chequeraImpresionCabeceraId;
	private Integer facultadId;
	private Integer tipochequeraId;
	private Long chequeraserieId;
	private Integer productoId;
	private Integer alternativaId;
	private Integer cuotaId;
	private String titulo = "";
	private BigDecimal total = BigDecimal.ZERO;
	private Integer cuotas = 0;
	private Integer mes = 0;
	private Integer anho = 0;
	private Integer arancelTipoId;

	@Column(name = "vencimiento_1")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime vencimiento1;

	@Column(name = "importe_1")
	private BigDecimal importe1 = BigDecimal.ZERO;

	@Column(name = "vencimiento_2")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime vencimiento2;

	@Column(name = "importe_2")
	private BigDecimal importe2 = BigDecimal.ZERO;

	@Column(name = "vencimiento_3")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime vencimiento3;

	@Column(name = "importe_3")
	private BigDecimal importe3 = BigDecimal.ZERO;

	private String usuarioId;

}

File: src/main/java/um/tesoreria/core/kotlin/model/ChequeraImpresionDetalle.kt
package um.tesoreria.core.kotlin.model

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import java.math.BigDecimal
import java.time.OffsetDateTime

@Entity
@Table(name = "chequera_d_impresion")
data class ChequeraImpresionDetalle(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var chequeraImpresionDetalleId: Long? = null,

    var chequeraImpresionCabeceraId: Long? = null,
    var facultadId: Int? = null,

    @Column(name = "tipochequera_id")
    var tipoChequeraId: Int? = null,

    @Column(name = "chequeraserie_id")
    var chequeraSerieId: Long? = null,

    var productoId: Int? = null,
    var alternativaId: Int? = null,
    var cuotaId: Int? = null,
    var titulo: String = "",
    var total: BigDecimal = BigDecimal.ZERO,
    var cuotas: Int = 0,
    var mes: Int = 0,
    var anho: Int = 0,
    var arancelTipoId: Int? = null,

    @Column(name = "vencimiento_1")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var vencimiento1: OffsetDateTime? = null,

    @Column(name = "importe_1")
    var importe1: BigDecimal = BigDecimal.ZERO,

    @Column(name = "vencimiento_2")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var vencimiento2: OffsetDateTime? = null,

    @Column(name = "importe_2")
    var importe2: BigDecimal = BigDecimal.ZERO,

    @Column(name = "vencimiento_3")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var vencimiento3: OffsetDateTime? = null,

    @Column(name = "importe_3")
    var importe3: BigDecimal = BigDecimal.ZERO,

    var usuarioId: String? = null,
    var version: Long? = null

) : Auditable()
